addons/popochiu/editor/gizmos/gizmo_clickable_plugin.gd

```diff
@@ -0,0 +1,241 @@
+@tool
+extends EditorPlugin
+
+# TODO: move these out of the plugin and into Popochiu (enums) or PopochiuClickable
+enum {
+	WALK_TO_POINT,
+	# LOOK_AT_POINT, # TODO: enable this when the look_at_point logic is implemented
+	BASELINE
+}
+
+# Private vars
+# State
+var _target_node: Node2D
+var _undo: EditorUndoRedoManager
+var _gizmos: Array
+var _grabbed_gizmo: Gizmo2D
+
+
+#region Virtual Methods implementations
+
+func _enter_tree() -> void:
+	# TODO: remove the following 2 lines when the plugin is connected to the appropriate signal
+	# e.g. popochiu_ready
+	PopochiuEditorConfig.initialize_editor_settings()
+	PopochiuConfig.initialize_project_settings()
+
+	# Initialization of the plugin goes here.
+	_undo = get_undo_redo()
+	_gizmos.insert(WALK_TO_POINT, _init_popochiu_gizmo(WALK_TO_POINT))
+	# TODO: enable this when the look_at_point logic is implemented
+	# _gizmos.insert(LOOK_AT_POINT, _init_popochiu_gizmo(LOOK_AT_POINT))
+	_gizmos.insert(BASELINE, _init_popochiu_gizmo(BASELINE))
+
+	EditorInterface.get_editor_settings().settings_changed.connect(_on_gizmo_settings_changed)
+
+
+func _exit_tree():
+	pass
+
+
+func _edit(object: Object) -> void:
```

Comment:
I didn't know about the existence of this virtual function. Now I've learned something new about the engine. Thanks!

The same applies for the `_handles()` one.